### DISCUSSION TITLE
Potential future development issues with IsotopicEnvelope class as result of deconvolution

### DISCUSSION THREAD

**Discussion by avcarr2** [smith-chem-wisc/mzLib#694] [Ideas]
I've defined two classes of deconvolution algorithms. The first is the THRASH-like algorithms that check the spacing between isotopic peaks to calculate charge state, find all isotopic peaks in a cluster, then calculate a mass. This is the automation of what you do when manually looking at spectra. The second class is true mathematical deconvolution, where you assume a peak in a spectra has a prior distribution that has been convoluted with another blurring distribution. Then you use an iterative method with a weighting function to recover the original distribution. This was the original, automated deconvolution method, which was implemented by Mann et al [1] in 1989. Maximum Entropy deconvolution and Michael Marty's UniDec are also examples of mathematical deconvolution. 

Here's the problem: If we want to implement mathematical deconvolution for low-resolution spectra (particularly top-down on TOFs) we will run into trouble in mzLib.  mzLib uses a THRASH-like deconvolution algorithm, and the output of deconvolution, IsotopicEnvelope, reflects that. Mathematical deconvolution will be unable to output the List of mz and intensity pairs that feed into a peak, a monoisotopic mass, or a charge. 

And since we want to implement mathematical deconvolution, we are unfortunately going to need to think about integrating two completely different types of results into the deconvolution workflow in mzLib. 

[1] Mann, Matthias, Chin Kai Meng, and John B. Fenn. "Interpreting mass spectra of multiply charged ions." Analytical Chemistry 61.15 (1989): 1702-1708.

**Comment by nbollis** [comment]
That is a good point and merits consideration. I think we are still okay to keep using the isotopic envelope as a deconvolution output for charge state deconvolution types. While the output of these algorithms is not an isotopic envelope, the only values we from the envelope for the search is neutral mass and charge state. Theoretically, the search would run just fine with empty peak mz and peak intensity arrays.
